gwcli/ceph.py

```diff
@@ -171,24 +149,92 @@ def ui_command_refresh(self):
         """
 
         self.refresh()
+        self.logger.info("ok")
+
+    def ui_command_info(self):
+
+        color = {"HEALTH_OK": "green",
+                 "HEALTH_WARN": "yellow",
+                 "HEALTH_ERR": "red"}
+
+        status = self.ceph_status
+        osdmap = status['osdmap']['osdmap']
+        output = "Cluster name: {}\n".format(self.cluster_name)
+        output += "Ceph version: {}\n".format(self.version)
+        output += "Health      : {}\n".format(self.health_status)
+        if self.health_status != 'HEALTH_OK':
+            output += " - {}\n".format(','.join(self.health_list))
+
+        mon_names = [mon.get('name') for mon in status["monmap"]["mons"]]
+
+        q_str = "quorum {}".format(','.join(status['quorum_names']))
+        q_out = set(mon_names) - set(status['quorum_names'])
+        if q_out:
+            q_str += ", out of quorum: {}".format(','.join(q_out))
+        output += "\nMONs : {:>4} ({})\n".format(self.topology.num_mons,
+                                                 q_str)
+        output += "OSDs : {:>4} ({} up, {} in)\n".format(self.topology.num_osds,
+                                                        osdmap['num_up_osds'],
+                                                        osdmap['num_in_osds'])
+        output += "Pools: {:>4}\n".format(self.num_pools)
+        raw = status['pgmap']['bytes_total']
+        output += "Raw capacity: {}\n".format(human_size(raw))
+        output += "\nConfig : {}\n".format(self.conf)
+        output += "Keyring: {}\n".format(os.path.join(CephGroup.ceph_config_dir,
+                                                      self.keyring))
+
+        console_message(output, color=color[self.health_status])
+
+    @property
+    def cluster_version(self):
+
+        vers_out = os_cmd("ceph -c {} version".format(self.conf))
```

Comment:
Why not just use the Python API's version method? [1]

[1] http://docs.ceph.com/docs/master/rados/api/python/#rados.Rados.version

Reply:
running ceph with -c to point to the appropriate cluster I get that clusters version. Maybe I'm doing something wrong, but this is what I get with the rados binding
>>> import rados
>>> r=rados.Rados(conffile='/etc/ceph/ceph.conf')
>>> str(r.version())
'0.69.1'

whereas this is what I'm after

e.g. 
/disks> /clusters/ceph/ info 
Cluster name: ceph
Ceph version: 12.1.3
Health      : HEALTH_OK

MONs :    3 (quorum ceph-1,ceph-2,ceph-3)
OSDs :    4 (4 up, 4 in)
Pools:    3
Raw capacity: 85G

Config : /etc/ceph/ceph.conf
Keyring: /etc/ceph/ceph.client.admin.keyring

/disks> exit
[root@rh7-gw1 ~]# ceph --version
ceph version 12.1.4 (a5f84b37668fc8e03165aaf5cbb380c78e4deba4) luminous (rc)

the gateway is on 12.1.4, but the cluster is running 12.1.3

Reply:
Hmm -- turns out that method just returns the constants defined in librados.h, which seems pointless. Disregard.